pyrobosim/pyrobosim/navigation/prm.py

```diff
@@ -41,18 +46,28 @@ def reset(self):
         self.latest_path = Path()
 
         # Create a search graph and sample nodes.
-        self.graph = SearchGraph(
-            world=self.world, max_edge_dist=self.max_connection_dist
-        )
+        self.graph = WorldGraph(color=[0, 0.4, 0.8], color_alpha=0.25)
+        self.graph.was_updated = True
         t_start = time.time()
         for i in range(self.max_nodes):
             n_sample = self.sample_configuration()
             if not n_sample:
                 warnings.warn(f"Could not sample more than {i} nodes")
                 break
-            self.graph.add(Node(n_sample), autoconnect=True)
+            self.graph.add_node(Node(pose=n_sample))
         self.sampling_time = time.time() - t_start
 
+        for node in self.graph.nodes:
+            self.connect_neighbors(node)
+
+    def connect_neighbors(self, node):
+        # Connect each node to all nodes within connection distance.
+        for other in self.graph.nodes:
+            if node == other:
+                continue
+            if self.is_connectable(node.pose, other.pose):
+                self.graph.add_edge(node, other)
+
     def plan(self, start, goal):
```

Comment:
During the planning phase of PRM in `plan()` method, the 'start` and `goal` nodes and edges from them should be removed after the planning is done.
Otherwise the graph keeps growing.